azurerm/resource_arm_databricks_workspace.go

```diff
@@ -187,7 +187,7 @@ func validateDatabricksWorkspaceName(v interface{}, k string) (ws []string, erro
 	}
 
 	// Cannot be more than 128 characters
-	if len(value) > 30 {
+	if len(value) > 128 {
```

Comment:
According to API spec, it should be between 3 and 64:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/databricks/resource-manager/Microsoft.Databricks/stable/2018-04-01/databricks.json#L684

I would suggest follow the API swagger.